dev_docs/cxg.md

```diff
@@ -73,38 +71,38 @@ For example:
 }
 ```
 
-### X
+### `X` array
 
-TileDB array, with a single anonymous attribute, shape (n_obs, n_var), containing the count matrix (equivalent to the AnnData `X` array).  MUST have numeric type, and SHOULD be float32.  The TileDB schema defines type and sparsity, and both dense and sparse encoding are supported.
+TileDB array, with a single anonymous attribute, shape `(n_obs, n_var)`, containing the count matrix (equivalent to the AnnData `X` array).  MUST have numeric type, and SHOULD be `float32`.  The TileDB schema defines type and sparsity, and both dense and sparse encoding are supported.
 
-### X_col_shift
+### `X_col_shift` array
 
-Optional TileDB array, used to encode-per column offsets for column-shift sparse encoding.  The TileDB array will have a single anonymous attribute, of the same type as the X array, and shape (n_var,).  
+Optional TileDB array, used to encode per-column offsets for column-shift sparse encoding.  The TileDB array will have a single anonymous attribute, of the same type as the X array, and shape `(n_var,)`.  
 
-If the X array is sparse, and X_col_shift exists, then all values in the i'th column were subtracted by X_col_shift[i].
+If the `X` array is sparse, and the `X_col_shift` array exists, then all values in the `i`'th column were subtracted by `X_col_shift[i]`.
 
-### emb and embedding arrays
+### `emb` group and embedding arrays
 
-A CXG must have a group named `emb`, which will contain all embeddings.  Embeddings are encoded as TileDB arrays, of numeric type and shape (n_obs, >=2).  The arrays SHOULD be coerced to float32, and MUST be a numeric type.  The TileDB array name will be assumed to be the embedding name (conventionally, embedding names in CXG are _not_ prefixed with an `X_` as they are in AnnData).
+A CXG MUST have a group named `emb`, which will contain all embeddings.  Embeddings are encoded as TileDB arrays, of numeric type and shape `(n_obs, >=2)`.  The arrays MUST be a numeric type, and SHOULD be coerced to `float32`. The array values MUST NOT contain any infinite values and MUST contain at least one non-NaN value.  The TileDB array name will be assumed to be the embedding name (conventionally, embedding names in CXG are _not_ prefixed with an `X_` as they are in AnnData).
```

Comment:
restate the numeric type requirement per the [CXG creation code](https://github.com/chanzuckerberg/single-cell-data-portal/blob/d9c534968d592aa9ecf46791b111e1e6edc061ba/backend/corpora/dataset_processing/h5ad_data_file.py#L129).

Reply:
The rewrite contains redundant "numeric type" requirements (one in the second sentence, another in the third).

I'm curious about your edits regarding non-scalars.  What is the use case for NaN's in an embedding?  Why not just say all embedding coordinates must have a finite value?

Reply:
The edits were only based upon what my reading of what the h5ad->cxg implementation is actually doing, rather than perhaps what it MUST be doing to make explorer happy.  I assumed the conversion code knew what it was doing and why 😄. So, yes, if explorer allows all float values, it doesn't seem worth documenting this as a spec constraint. Though it seems cleaner to insist on finite floats for the intended visualization purpose of these embeddings.

Reply:
I think the real question is if there is any viable use case for an embedding algo to generate a non-finite.  i can't think of one, although I recommend validating with Amborse/Pablo/Jason.  Assuming I am correct, then I'd tighten this to "only finite float values"

Reply:
taking this to [slack](https://czi-sci.slack.com/archives/C024HCSH9PT/p1631579063117000)

Reply:
TODO: I will relax this to state what is actually allowed by the explorer code, and if we tighten the constraints, we'll want to bump  CXG version along with the Explorer code changes.